proposals/stdlib/locale-agnostic-string-conversions.md

```diff
@@ -0,0 +1,139 @@
+>Note: This method is locale sensitive, and may produce unexpected results if used for strings that are intended to be interpreted locale independently. 
+>Examples are programming language identifiers, protocol keys, and HTML tags. For instance, "title".toUpperCase() in a Turkish locale returns "T\u0130TLE", 
+>where '\u0130' is the LATIN CAPITAL LETTER I WITH DOT ABOVE character. To obtain correct results for locale insensitive strings, use toUpperCase(Locale.ROOT).
+
+Bug report related to the caution: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6208680 
+
+However, documentation doesn't always prevent misuse. See bug reports related to this issue in big projects:
+* https://issues.apache.org/jira/browse/SPARK-20156
+* https://github.com/gradle/gradle/issues/1506
+
+Related blog posts:
+* https://lotusnotus.com/lotusnotus_en.nsf/dx/dotless-i-tolowercase-and-touppercase-functions-use-responsibly.htm
+* https://javapapers.com/core-java/javas-tolowercase-has-got-a-surprise-for-you/
+
+To combat the issue we would like to deprecate current API and introduce new locale-agnostic functions.
+
+## Description
+
+* Deprecate existing locale-dependent function `String.toUpperCase(): String` and introduce locale-agnostic function `String.uppercase(): String` as a replacement 
+* Deprecate existing locale-dependent function `String.toLowerCase(): String` and introduce locale-agnostic function `String.lowercase(): String` as a replacement 
+* Deprecate existing locale-dependent function `String.capitalize(): String` and introduce locale-agnostic function `String.capitalizeFirst(): String` as a replacement 
+* Deprecate existing locale-dependent function `String.decapitalize(): String` and introduce locale-agnostic function `String.decapitalizeFirst(): String` as a replacement 
+
+```kotlin
+/**
+ * Returns a copy of this string converted to upper case using Unicode mapping rules of the invariant locale.
+ */
+public expect fun String.uppercase(): String
+
+/**
+ * Returns a copy of this string converted to lower case using Unicode mapping rules of the invariant locale.
+ */
+public expect fun String.lowercase(): String
+
+/**
+ * Returns a copy of this string having its first letter titlecased using Unicode mapping rules of the invariant locale,
+ * or the original string if it's empty or already starts with a title case letter.
+ */
+public expect fun String.capitalizeFirst(): String
+
+/**
+ * Returns a copy of this string having its first letter lowercased using Unicode mapping rules of the invariant locale,
+ * or the original string if it's empty or already starts with a lower case letter.
+ */
+public expect fun String.decapitalizeFirst(): String
+```
+
+The new namings will also affect following functions:
+
+* Rename `String.toLowerCase(locale: Locale): String` to `String.lowercase(locale: Locale): String`
```

Comment:
Here "Rename" makes an impression of something completely different from "deprecate" + "introduce new name".

Reply:
Rephrased